### ISSUE TITLE
[TODO] SPMS API

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/SPMS_ERP_MRPUpdatedRepo#24]
Two components are calling for spmsAPI, but are defined and never used.

These files are EmployeeList.jsx and EmployeeDetail.jsx in /src/client/

Right now, spms-erpi.js API in /src/client/apis/ only using Axios.

There is also another pair of functions calling the API, auth.service.js and employee.service.js

**Comment by TechLateef** [comment]
if i may ask what is need to be done here

**Comment by gbowne1** [comment]
I do not think this part is working.  Its been a lomg time though. @TechLateef

**Comment by TechLateef** [comment]
Yes I have seen some issues I was about doing that when my boss called me for some unfinished work I will try and make some fix tomorrow I've seen some

**Comment by gbowne1** [comment]
@TechLateef no worries at all.  It's been around 4 years since I last worked on this other than trying to update it to the latest packages.. etc. 

the spms_erpmrp dir can be removed without concern.

everything is in /src

**Comment by TechLateef** [comment]
Alright then

**Comment by gbowne1** [comment]
I missed pushing up the last round of updates.. so I will do that today. @TechLateef

**Comment by TechLateef** [comment]
Alright

**Comment by TechLateef** [comment]
from what i see in this project the signUp and signin are only for the admin non was created for the employee and again the signUp as some issues but am trying to figure out the problem 

so do you need an endpoint for the employees ?

**Comment by gbowne1** [comment]
Yes. @TechLateef we started with all the HR and employee stuff but the person helping me gave up as neither of us knew much about the react or the rest of the stack

**Comment by TechLateef** [comment]
am talking of the back-end here

i dont do front end 

and again am not familiar with docker one thing i will be needing now do you have a postgresql account i will need some infor so i can connect it with the app

**Comment by TechLateef** [comment]
user: 'sgpostgres',
  host: 'SG-PostgreNoSSL-14-pgsql-master.devservers.scalegrid.io',
  database: 'postgres',
  password: 'password',
  port: 5432,

for the db

**Comment by gbowne1** [comment]
@TechLateef It was all runnning inside the docker.  I did not do the backend myself.  I do have PostgreSQL locally though. 

I do have the original usernames and passwords.  But I can connect to any db in VSCode.

**Comment by TechLateef** [comment]
you cant connect it on VSCode?


okay you know what let me go through the Docker documentation and play with it a little bit and see

**Comment by gbowne1** [comment]
ok.

**Comment by TechLateef** [comment]
how do you access the app locally?? after running it on container

**Comment by gbowne1** [comment]
@TechLateef 

I only have ran the app locally through the create react app development server... npm start in both the client and the server.

I don't know a lot about Docker so I never did that workflow, with it being as I mainly worked on the Frontend.

**Comment by TechLateef** [comment]
When you do so have you interact with the backend like signup for admin ??

**Comment by gbowne1** [comment]
@TechLateef I have PostgreSQL and had it running locally.      My original plan was to host it on heroku postgres hosting  or Amazon AWS RD but you can only have so much data on the free plan and most of them only let you have 30 days. If we can keep test data under 3gbk that might be possible

**Comment by TechLateef** [comment]
@gbowne1  my question here is have you integrate the frontend with the backend?

**Comment by gbowne1** [comment]
@TechLateef No, because at that point I had stopped to work on other things. I was only working on the frontend.  I did not do any of the integration.

**Comment by gbowne1** [comment]
Which ones?

Both /src/client and /src/server must be installed as both have package.json and their own package dependencies

**Comment by TechLateef** [comment]
file:///home/techlateef/Pictures/Screenshots/Screenshot%20from%202023-03-09%2008-55-29.png

**Comment by gbowne1** [comment]
I believe the login values were hard coded. Its been a long time.  

The server and nodemon appear to start fine but the frontend is not starting because there are so many errors.

I tried starting it again the other day.  Its just a blank screen covered by eslint errors.

Any other issues it has I don't know.

**Comment by gbowne1** [comment]
@TechLateef screen shot did not render.  Try again.

**Comment by TechLateef** [comment]
code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: mui-datatables@4.2.2
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR!   peer react@">=16.8.0" from @emotion/react@11.10.0
npm ERR!   node_modules/@emotion/react
npm ERR!     @emotion/react@"^11.10.0" from the root project
npm ERR!     peer @emotion/react@"^11.0.0-rc.0" from @emotion/styled@11.10.0
npm ERR!     node_modules/@emotion/styled
npm ERR!       @emotion/styled@"^11.10.0" from the root project
npm ERR!       4 more (@mui/lab, @mui/material, @mui/styled-engine, @mui/system)
npm ERR!     6 more (@mui/lab, @mui/material, @mui/styled-engine, ...)
npm ERR!   20 more (@emotion/styled, @fortawesome/react-fontawesome, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.2" from mui-datatables@4.2.2
npm ERR! node_modules/mui-datatables
npm ERR!   mui-datatables@"^4.2.2" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peer react@"^16.8.0 || ^17.0.2" from mui-datatables@4.2.2
npm ERR!   node_modules/mui-datatables
npm ERR!     mui-datatables@"^4.2.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/techlateef/.npm/_logs/2023-03-09T08_53_37_473Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/techlateef/.npm/_logs/2023-03-09T08_53_37_473Z-debug-0.log

**Comment by gbowne1** [comment]
You might try `npm install --force --legacy-peer-deps`

I had the same issue.

**Comment by TechLateef** [comment]
This didn't work but I make I new pill yesterday and there's some problem with your Frontend code

**Comment by gbowne1** [comment]
@TechLateef 

Yes there is lots of problems with the frontend.  This is why I attempted to fix the front end in the folder labeled spms_erpmrp.

At this point, I am not sure what to do.

**Comment by TechLateef** [comment]
Alright I will hold on for now when you're kinda done let me know 

Because I see you like finish with your front end before backend
